wandb/sdk/wandb_settings.py

```diff
@@ -434,6 +440,10 @@ def _infer_settings_from_env(self):
                 console = "off"
             u["console"] = console
 
+        # convert wandb mode to "offline"
+        if self.mode == "dryrun":
+            self.offline = True
```

Comment:
Does this mess up the format_dict for offline since "run_mode" could be "run" at this point?

Reply:
this is early enough, this is why we have the early logger, so we can do all the setup stuff first and in the final part of wandb.init() setup, generate the log files.